src/main/scala/org/broadinstitute/dsde/firecloud/service/WorkspaceService.scala

```diff
@@ -24,20 +26,22 @@ trait WorkspaceService extends HttpService with PerRequestCreator with FireCloud
   lazy val log = LoggerFactory.getLogger(getClass)
   lazy val rawlsWorkspacesRoot = FireCloudConfig.Rawls.workspacesUrl
 
+  def transformSingleWorkspaceRequest(entity: HttpEntity): HttpEntity = {
+    entity.as[RawlsWorkspaceResponse] match {
+      case Right(rwr) => new UIWorkspaceResponse(rwr).toJson.prettyPrint
```

Comment:
entity.as[T] is returning an Either.  I agree that it's kinda weird, do you know a better way to attempt the unmarshalling?

And as-is, it's not liking just `.toJson`, it's needing it as a string in order to convert back to HttpEntity.  Maybe there's more implicit magic I'm missing.